### PR TITLE
audio: volume: Fix compilation error when COM_PEAK_VOL is not selected

### PR DESCRIPTION
**Pull request by ranj063**
CONFIG_PEAK_METER_UPDATE_PERIOD depends on CONFIG_COMP_PEAK_VOL

### CODE REVIEW

File: src/audio/module_adapter/module/volume/volume.c [hunk 1]
```diff
@@ -1294,7 +1294,9 @@ static int volume_prepare(struct processing_module *mod,
 
 #if CONFIG_IPC_MAJOR_4
 	cd->peak_cnt = 0;
+#if CONFIG_COMP_PEAK_VOL
 	cd->peak_report_cnt = CONFIG_PEAK_METER_UPDATE_PERIOD * 1000 / mod->dev->period;
+#endif
```
Comment (btian1) [inline src/audio/module_adapter/module/volume/volume.c:1299]: #endif should goto line 1302?

Comment (ranj063) [reply src/audio/module_adapter/module/volume/volume.c:1299]: I'm not sure about this. Maybe all of peak_cnt reporting should go within the #if. But this is enough for now to fix the compilation error when the config isn't enabled

Comment (kv2019i) [reply src/audio/module_adapter/module/volume/volume.c:1299]: I'd say 1302 needs to be included as well.

s/COM/COMP/ typo also in cmmit title

**Comment by andrula-song** [comment]
then why not fix the definition of struct vol_data in volume.h as well?

**Comment by ranj063** [comment]
> then why not fix the definition of struct vol_data in volume.h as well? line 145 should be "#if CONFIG_COMP_PEAK_VOL" too.

@andrula-song I think those are fine to be left in. THey will be unused if PEAK_VOL isnt selected

**Review by btian1 — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@wszypelt @lrudyX think this may be a false positove as this is fixing a compilation error with volume. Good to merge ?

**Comment by wszypelt** [comment]
@lgirdwood good to merge :)

**Pull request merged**